### PR TITLE
Expand MailChimp Sync

### PR DESCRIPTION
**Pull request by patrickvenetz**
This Pull Requests updates first and last name merge fields (`FNAME`, `LNAME`) on MailChimp whenever there is a chance of a change, as well as email base64 url-safe encoded merge field (`EMAILBASE64U`).

It also helps updating MailChimp data in batches. You can run individual "operation types" with

```
$ packages/mail/script/mailchimp/runOperations.js --type nameAndEmailBase64u
```

Operation types can be extended in packages/mail/lib/operations and available thus far are:
- `covidAccessToken` to updated Authorize Session Access Tokens in lieu of COVID-19 newsletter
- `nameAndEmailBase64u` to updated name and email base64u fields

Batches can take several hours to complete. Run `runOperations.js --no-interval` to end script once batch is submitted and a batch can be checked manually:

```
$ packages/mail/script/mailchimp/checkBatch.js --id 123123123
```

### CODE REVIEW

File: packages/mail/script/mailchimp/runOperations.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            { batch: id, status, submittedAt, totalOperations, finishedOperations }
+          )
+
+          isRunning = status !== 'finished'
+
+          if (isRunning) {
+            await Promise.delay(1000 * argv.interval)
+          }
+        } while (isRunning)
+      } else { // if (argv.interval) {
```
Comment (patte) [inline packages/mail/script/mailchimp/runOperations.js:75]:
```suggestion
      } else { // if (!argv.interval) {
```

File: packages/mail/MailchimpInterface.js [hunk 2]
```diff
@@ -89,6 +110,34 @@ const MailchimpInterface = ({ logger }) => {
...
+        return {
+          ...rest,
+
+          // stringify body if not a sstring just yet
+          ...typeof operation.body !== 'string' && { body: JSON.stringify(operation.body) },
+
+          // when subscriber hash is provided, overwrite path to member resource
+          ...subscriberHash && {
+            path: `/lists/${MAILCHIMP_MAIN_LIST_ID}/members/${subscriberHash}`
+          }
```
Comment (patte) [inline packages/mail/MailchimpInterface.js:127]: did this speed up the batch process?

Comment (patrickvenetz) [reply packages/mail/MailchimpInterface.js:127]: No.

File: packages/mail/script/mailchimp/runOperations.js (hunk 1 above)
Comment (patrickvenetz) [reply packages/mail/script/mailchimp/runOperations.js:75]: Won't change. This comment shows which if-clause it belongs to.

File: packages/mail/lib/operations/covidAccessTokens.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+module.exports = async ({ pgdb }) => {
+  const users = await pgdb.query(`
+    WITH "grantsAndRevokes" AS (
+      SELECT c."userId"
+      FROM "consents" c
+      WHERE c.policy = 'NEWSLETTER_COVID19'
+    )
+    
+    SELECT u.*, md5(lower(u.email)) "__subscriberHash", COUNT(*) % 2 > 0 "__subscribed"
```
[the remark this answers is not in the document]
Comment (patrickvenetz) [reply packages/mail/lib/operations/covidAccessTokens.js]: Adopted.

**Comment by patrickvenetz** [comment]
@patte Please check this one again, too. Muchos gracias.

File: packages/mail/lib/operations/nameAndEmailBase64u.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ module.exports = async ({ pgdb }) => {
   const users = await pgdb.query(`
     SELECT u."firstName", u."lastName", md5(lower(u.email)) "__subscriberHash"
     FROM "users" u
-    WHERE u.verified = TRUE
+    WHERE u.verified = TRUE AND u."deletedAt" IS NULL
```
Comment (patte) [inline packages/mail/lib/operations/nameAndEmailBase64u.js:9]: good catch!

**Pull request merged**